pkgs/by-name/st/starpls/package.nix

```diff
@@ -14,11 +14,7 @@ stdenv.mkDerivation (finalAttrs: {
   pname = "starpls";
   version = manifest.version;
 
-  src =
-    let
-      system = stdenv.hostPlatform.system;
-    in
-    fetchurl (manifest.assets.${system} or (throw "Unsupported system: ${system}"));
+  src = fetchurl (manifest.assets.${stdenv.hostPlatform.system});
```

Comment:
Why remove this?

Reply:
It’s not needed here, meta.platforms should work.

Reply:
`meta.platforms` is moreso a suggestion that can be easily overridden (and often is) by `$NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM` or the `allowUnsupportedSystem` config option. Having fallbacks like these when accessing attrsets in easily fallible ways helps avoid extremely vague error messages that can result from that, and should definitely be kept